pkg/services/sqlstore/dashboard_test.go

```diff
@@ -259,6 +259,35 @@ func TestDashboardDataAccess(t *testing.T) {
 				So(hit.FolderTitle, ShouldEqual, "")
 			})
 
+			Convey("Should be able to limit search", func() {
+				query := search.FindPersistedDashboardsQuery{
+					OrgId:        1,
+					Limit:        1,
+					SignedInUser: &m.SignedInUser{OrgId: 1, OrgRole: m.ROLE_EDITOR},
+				}
+
+				err := SearchDashboards(&query)
+				So(err, ShouldBeNil)
+
+				So(len(query.Result), ShouldEqual, 1)
+				So(query.Result[0].Title, ShouldEqual, "1 test dash folder")
+			})
+
+			Convey("Should be able to search beyond limit using paging", func() {
+				query := search.FindPersistedDashboardsQuery{
+					OrgId:        1,
+					Limit:        1,
+					Page:         2,
+					SignedInUser: &m.SignedInUser{OrgId: 1, OrgRole: m.ROLE_EDITOR},
+				}
+
+				err := SearchDashboards(&query)
+				So(err, ShouldBeNil)
+
+				So(len(query.Result), ShouldEqual, 1)
+				So(query.Result[0].Title, ShouldEqual, "1 test dash folder")
```

Comment:
Shouldn't this be a dashboard? Or is folder always included in search result if there are a dashboard? Think we need a few more tests in that case to cover the case when both folder and dashboard are expected to be returned.

Reply:
Search returns both folders & dashboards, no?

Reply:
Sorry for not being clear. 

The test "Should be able to limit search" are using page=1 and limit=1 and you verify that you get "1 test dash folder" in response. However in the test highlighted above "Should be able to search beyond limit using paging" you also verify that you get "1 test dash folder" in response where I think that should be something else since you're asking for page 2 with a limit of 1.

Reply:
ok, that is strange. Investigated this revealed the limit is not working, should not be on the outer query but the inner. As the outer is joining with tags.

Reply:
Great :+1: Awaiting a fix then